spec/features/consumer/account_spec.rb

```diff
@@ -70,7 +70,7 @@
 
         # It reveals table of orders for distributors when clicked
         expand_active_table_node distributor1.name
-        expect(page).to have_link "Order " + d1o1.number, href: "/orders/#{d1o1.number}"
+        expect(page).to have_link "Order " + d1o1.number, href: "/spree/orders/#{d1o1.number}"
```

Comment:
this is really the only issue with this PR. The actual URL will be changed.
I wonder if this is a problem. I wonder if there are links in emails that could be pointing at the previous URL.
I also wonder if there is a quick way around and keep the url as it was but still make this change in the routes...
@Matt-Yorkley  any idea?

Reply:
> I wonder if there are links in emails that could be pointing at the previous URL.

Not only that but also any link could have been saved, shared, bookmarked and things will break. So IMO, if we can avoid this breaking change, better.

We can do that by using [scope](https://apidock.com/rails/v4.2.9/ActionDispatch/Routing/Mapper/Scoping/scope) (see  _2.6 Controller Namespaces and Routing_ on https://guides.rubyonrails.org/v4.2/routing.html#controller-namespaces-and-routing). This results in the following:
```
diff --git a/app/serializers/api/order_serializer.rb b/app/serializers/api/order_serializer.rb
index 0f7cc4dde..85937779d 100644
--- a/app/serializers/api/order_serializer.rb
+++ b/app/serializers/api/order_serializer.rb
@@ -42,7 +42,7 @@ module Api
     end
 
     def path
-      spree_order_path(object)
+      order_path(object)
     end
 
     def cancel_path
diff --git a/config/routes/spree.rb b/config/routes/spree.rb
index 7a49520f0..91215dce9 100644
--- a/config/routes/spree.rb
+++ b/config/routes/spree.rb
@@ -1,5 +1,5 @@
 Openfoodnetwork::Application.routes.draw do
-  namespace :spree do
+  scope module: 'spree' do
     resources :orders do
       get :clear, :on => :collection
       get :order_cycle_expired, :on => :collection
diff --git a/spec/features/consumer/account_spec.rb b/spec/features/consumer/account_spec.rb
index c9e55dfd5..7011c0c1b 100644
--- a/spec/features/consumer/account_spec.rb
+++ b/spec/features/consumer/account_spec.rb
@@ -70,7 +70,7 @@ feature '
 
         # It reveals table of orders for distributors when clicked
         expand_active_table_node distributor1.name
-        expect(page).to have_link "Order " + d1o1.number, href: "/spree/orders/#{d1o1.number}"
+        expect(page).to have_link "Order " + d1o1.number, href: "/orders/#{d1o1.number}"
 
         expand_active_table_node distributor2.name
         expect(page).not_to have_content "Order " + d1o1.number.to_s
```

And the routes end up like:

```
              clear_orders GET    /orders/clear(.:format)                                                          spree/orders#clear          
order_cycle_expired_orders GET    /orders/order_cycle_expired(.:format)                                            spree/orders#order_cycle_expired
              cancel_order PUT    /orders/:id/cancel(.:format)                                                     spree/orders#cancel          
                    orders GET    /orders(.:format)                                                                spree/orders#index             
                           POST   /orders(.:format)                                                                spree/orders#create            
                 new_order GET    /orders/new(.:format)                                                            spree/orders#new                
                edit_order GET    /orders/:id/edit(.:format)                                                       spree/orders#edit
                     order GET    /orders/:id(.:format)                                                            spree/orders#show
                           PATCH  /orders/:id(.:format)                                                            spree/orders#update
                           PUT    /orders/:id(.:format)                                                            spree/orders#update
                           DELETE /orders/:id(.:format)                                                            spree/orders#destroy
```

Reply:
That's exactly what we need :ok_hand:

Reply:
nice one Pau. It just works!